stixcore/processing/pipeline.py

```diff
@@ -50,36 +58,163 @@ def __init__(self, func, regex):
         if not isinstance(regex, type(re.compile('.'))):
             raise TypeError('regex must be a regex Pattern')
         self.regex = regex
+        self.args = args
+        self.queue = Queue(maxsize=0)
+        self.name = name
+        self.lp = threading.Thread(target=self.process)
+        self.lp.start()
+
+    @poll_decorator(step=1, poll_forever=True)
```

Comment:
So you poll but only will the `queue.get` not wait forever at each poll?

Reply:
yes the  queue.get will wait until something new comes in.
But after that the process is done.
the poll_decorator starts a new process afterwards again and again .

Reply:
What the advantage compare to doing something like this inside the function? 
```
while True:
    path = self.queue.get()
    try:
         self.func(path, **self.args)
    except Exception as e:
        logger.error(e)
             if CONFIG.getboolean('Logging', 'stop_on_error', fallback=False):
                 raise e